src/lib/form-field/form-field.scss

```diff
@@ -144,6 +144,8 @@ $mat-form-field-default-infix-width: 180px !default;
   width: 100%;
   // Need this so that the underline doesn't block the hover effect.
   pointer-events: none;
+  // We transform the height slightly higher to fix inconsistent underline height for some DPIs.
+  transform: scaleY(1.0001);
```

Comment:
Are you able to elaborate on _why_ adding this scale resolves the issue?

Reply:
At zoom levels between 50% and 100% some of the form-fields would lose their underline and some would have `1px` underline. It seemec related to fractional pixels in some way (e.g. underlines with their top at `x.7px` were fine, but underlines at `x.8px` disappeared). The exact fractions that worked or didn't work seemed to depend on the users resolution and zoom level.

This was basically just a hunch "hey maybe making the underline barely thicker will somehow help." And on all the screens and zoom levels I tested it did indeed fix the problem. It also fixed the case where you're zoomed in to >100% and the underlines would have inconsistent thickness.

Reply:
Add the gist of that to the comment?